core/trino-main/src/main/java/io/trino/connector/system/jdbc/FilterUtil.java

```diff
@@ -50,6 +50,20 @@ public static QualifiedTablePrefix tablePrefix(String catalog, Optional<String>
         return new QualifiedTablePrefix(catalog);
     }
 
+    public static Optional<QualifiedTablePrefix> tablePrefixIfPossible(String catalog, Optional<String> schema, Optional<String> table)
+    {
+        if (schema.isPresent() && table.isPresent()) {
+            return Optional.of(new QualifiedTablePrefix(catalog, schema.get(), table.get()));
+        }
+        if (schema.isPresent() && table.isEmpty()) {
```

Comment:
my IDE warns me: "Condition 'table.isEmpty()' is always 'true' when reached "

Reply:
Yeah - that is true. It felt more readable to me to keep it still in condition so we explicitly say what value of schema/table we are expecting.
Can change if you prefer, not a big deal.